Mathlib/MeasureTheory/Group/Action.lean

```diff
@@ -103,6 +104,36 @@ theorem map_smul : map (c • ·) μ = μ :=
 
 end MeasurableSMul
 
+section SMulHomClass
+
+universe uM uN uα uβ
+variable {M : Type uM} {N : Type uN}  {α : Type uα} {β : Type uβ}
+  [MeasurableSpace M] [MeasurableSpace N] [MeasurableSpace α] [MeasurableSpace β]
+
+@[to_additive]
+theorem SMulInvariantMeasure_map [SMul M α] [SMul M β]
+    [MeasurableSMul M β]
+    (μ : Measure α) [SMulInvariantMeasure M α μ] (f : α → β)
+    (hsmul : ∀ (m : M) a, f (m • a) = m • f a) (hf : Measurable f) :
+    SMulInvariantMeasure M β (map f μ) where
+  measure_preimage_smul m S hS := calc
+    map f μ ((m • ·) ⁻¹' S)
+    _ = μ (f ⁻¹' ((m • ·) ⁻¹' S)) := map_apply hf <| hS.preimage (measurable_const_smul _)
+    _ = μ ((m • f ·) ⁻¹' S) := by rw [preimage_preimage]
+    _ = μ ((f <| m • ·) ⁻¹' S) := by simp_rw [hsmul]
+    _ = μ ((m • ·) ⁻¹' (f ⁻¹' S)) := by rw [←preimage_preimage]
+    _ = μ (f ⁻¹' S) := by rw [SMulInvariantMeasure.measure_preimage_smul m (hS.preimage hf)]
+    _ = map f μ S  := (map_apply hf hS).symm
```

Comment:
I think this could be made slightly nicer by *not* picking a set `S` and just using the functorial properties of `map`. That seems to require an extra `MeasurableSMul` assumption but I think this is completely harmless. However we would need a bit more API, so I'll just do it in a separate PR once this is merged.